docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+
+## Container variables to adjust
+
+Since Docker Mailserver provides minimal support for Lua with Dovecot, there are no environment variables to configure specifically for Lua. Some environment variables that do exist for other aspects of DMS might need to be configured based on the scenario you want to follow.
+
+In the case of the example scenario, the environment variables must be configured as if users will be authenticated against LDAP (to support identification of mailboxes and verifying authorizations through LDAP). See [LDAP Authentication][docs-authldap] for more information. In addition, the following variables are required:
+
+???+ example
+
+    DMS configuration environment variables to let Postfix apply Lua authentication by proxy through Dovecot for authenticated SMTP.
+
+    ```yaml
+    - ENABLE_SASLAUTHD=1
+    - SASLAUTHD_MECHANISMS=rimap
+    - SASLAUTHD_MECH_OPTIONS=127.0.0.1
+    ```
+
+## Modifying Dovecot's configuration
+
+Add the following volume values to the relevant part of `compose.yaml`:
+
+???+ example
+
+    Override and add Dovecot configuration files and lua scripts.
+
+    ```yaml
+        volumes:
+          - ./docker-data/dms/config/dovecot/auth-ldap.conf.ext:/etc/dovecot/conf.d/auth-ldap.conf.ext:ro
+          - ./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf:/etc/dovecot/conf.d/auth-lua-httpbasic.conf:ro
+          - ./docker-data/dms/config/dovecot/lua/:/etc/dovecot/lua/:ro
+    ```
+
+The first volume line [overrides][docs-dovecotoverrideconfiguration] Dovecot's standard LDAP authentication configuration file. The second line [adds][docs-dovecotaddconfiguration] a new configuration file for Lua authentication. The third line adds a directory which will contain Lua scripts. The files and directory will not be changed from inside the container, which is why they are configured as read-only.
+
+Make the necessary changes on the filesystem (*where `mailserver` is the container name*):
+```bash
+mkdir -p ./docker-data/dms/config/dovecot/lua
+docker cp mailserver:/etc/dovecot/conf.d/auth-ldap.conf.ext ./docker-data/dms/config/dovecot/auth-ldap.conf.ext
+```
+
+Edit configuration file `./docker-data/dms/config/dovecot/auth-ldap.conf.ext`. Comment out the passdb section. An excerpt of what that part would look like after you are done:
+```
+#passdb {
+#  driver = ldap
+#
+#  # Path for LDAP configuration file, see example-config/dovecot-ldap.conf.ext
+#  args = /etc/dovecot/dovecot-ldap.conf.ext
+#}
+```
+
+Don't touch anything else in the file.
+
+Create configuration file `./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf` with contents:
+```
+passdb {
+  driver = lua
+  args = file=/etc/dovecot/lua/auth-httpbasic.lua blocking=yes
+}
+```
```

Comment:
Likewise, this may break for you in future when you're relying on DMS being configured for a specific `ACCOUNT_PROVISIONER` which makes certain assumptions. This is probably another caveat that could be mentioned in the disclaimer at the start of this guide, since breakage can work both ways.

When the OIDC feature is closer to being merged, we may be able to support an BYO config, but I think that'd still run into compatibility issues with some features of DMS (_eg: Dovecot sieve scripts, v13 recently received a fix to adjust the home directory within an accounts mailbox, LDAP doesn't have that and some users may explicitly configure their own location which won't be compatible_). Semver releases will hopefully at least prevent the likelihood of breakage for you between major releases.

---

BTW this section for Dovecot seems it's missing enabling the config? The LDAP config will be included, but you'll likely need a line similar to this:

https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/dovecot/10-auth.conf#L125

I don't recommend volume mounting a replacement config for `10-auth.conf`, so `user-patches.sh` is probably beneficial for you. Keep in mind that if you just append the needed line, without a conditional to prevent adding it again this will pile up across container restarts (_unless the user follows our advice to always re-create the container, but many seem to miss this_), as internal state is persisted within the container until it's removed.

Eventually, we'll probably provide our customizations from a separate location, and always copy them over during container startup to minimize the issue so that config modifications are more deterministic.

Reply:
> Likewise, this may break for you in future when you're relying on DMS being configured for a specific `ACCOUNT_PROVISIONER` which makes certain assumptions. This is probably another caveat that could be mentioned in the disclaimer at the start of this guide, since breakage can work both ways.

A good point. I added an additional line to the warning at the top of this guide to note this.

> When the OIDC feature is closer to being merged, we may be able to support an BYO config, but I think that'd still run into compatibility issues with some features of DMS (_eg: Dovecot sieve scripts, v13 recently received a fix to adjust the home directory within an accounts mailbox, LDAP doesn't have that and some users may explicitly configure their own location which won't be compatible_). Semver releases will hopefully at least prevent the likelihood of breakage for you between major releases.

We'll cross that bridge when we get there.

> BTW this section for Dovecot seems it's missing enabling the config? The LDAP config will be included, but you'll likely need a line similar to this:
> 
> https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/dovecot/10-auth.conf#L125
> 
> I don't recommend volume mounting a replacement config for `10-auth.conf`, so `user-patches.sh` is probably beneficial for you. Keep in mind that if you just append the needed line, without a conditional to prevent adding it again this will pile up across container restarts (_unless the user follows our advice to always re-create the container, but many seem to miss this_), as internal state is persisted within the container until it's removed.

Unnecessary. `auth-lua-httpbasic.conf` has the `.conf` extension. All `.conf` files are loaded automatically. See:

```
dovecot.conf:!include conf.d/*.conf
```

> Eventually, we'll probably provide our customizations from a separate location, and always copy them over during container startup to minimize the issue so that config modifications are more deterministic.

I am looking forward to that. Please document this, and point me in the right direction when the time comes. I will be happy to adjust the current example to a better supported environment for community contributions.